src/Evolution/Systems/ForceFree/RecoverPhysicalVariables.hpp

```diff
@@ -0,0 +1,90 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include "DataStructures/DataBox/Tag.hpp"
+#include "DataStructures/Tensor/TypeAliases.hpp"
+#include "Evolution/Systems/ForceFree/Tags.hpp"
+#include "PointwiseFunctions/GeneralRelativity/TagsDeclarations.hpp"
+#include "Utilities/TMPL.hpp"
+
+/// \cond
+class DataVector;
+namespace gsl {
+template <typename T>
+class not_null;
+}  // namespace gsl
+/// \endcond
+
+namespace ForceFree::Tags {
+/*!
+ * \brief Compute the electric field \f$E^i\f$ from TildeE.
+ *
+ * \note This ComputeTag is solely for observation purpose, not related to
+ * actual time evolution.
+ */
+struct ElectricFieldCompute : ElectricField, db::ComputeTag {
+  using argument_tags =
+      tmpl::list<TildeE, gr::Tags::SqrtDetSpatialMetric<DataVector>>;
+  using return_type = tnsr::I<DataVector, 3>;
+  using base = ElectricField;
+
+  static void function(gsl::not_null<tnsr::I<DataVector, 3>*> electric_field,
```

Comment:
Could you move these into free functions please, and just reference in the compute item?

Reply:
I'll try that